### PR TITLE
feat!: Add support for Optimum optimizers and quantizers

### PR DESCRIPTION
**Pull request by shadeMe**
This PR adds support for Optimum-backed optimization and quantization of models.

### CODE REVIEW

File: integrations/optimum/src/haystack_integrations/components/embedders/optimum/quantization.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    support by the Optimum Embedders.
+    """
+
+    #: Quantization for the ARM64 architecture.
+    ARM64 = "arm64"
+
+    #: Quantization with AVX-2 instructions.
+    AVX2 = "avx2"
+
+    #: Quantization with AVX-2 instructions.
```
Comment (sjrl) [inline integrations/optimum/src/haystack_integrations/components/embedders/optimum/quantization.py:20]: I think you meant `AVX-512 instructions` here

File: integrations/optimum/src/haystack_integrations/components/embedders/optimum/optimum_text_embedder.py [hunk 2]
```diff
@@ -50,41 +41,62 @@ def __init__(
...
+             In case of duplication, these kwargs override `model`, `onnx_execution_provider`
+             and `token` initialization parameters.
+         :param working_dir:
+             The directory to use for storing intermediate files
+             generated during model optimization/quantization.
+
+             Required for optimization and quantization.
+         :param optimizer_settings:
+             Configuration for Optimum Embedder Optimization.
+             If `None`, no additional optimization is be applied.
```
Comment (silvanocerza) [inline integrations/optimum/src/haystack_integrations/components/embedders/optimum/optimum_text_embedder.py:96]:
```suggestion
             If `None`, no additional optimization is applied.
```

File: integrations/optimum/src/haystack_integrations/components/embedders/optimum/optimum_text_embedder.py [hunk 3]
```diff
@@ -50,41 +41,62 @@ def __init__(
...
+         :param quantizer_settings:
+             Configuration for Optimum Embedder Quantization.
+             If `None`, no quantization is be applied.
```
Comment (silvanocerza) [inline integrations/optimum/src/haystack_integrations/components/embedders/optimum/optimum_text_embedder.py:99]:
```suggestion
             If `None`, no quantization is applied.
```

File: integrations/optimum/src/haystack_integrations/components/embedders/optimum/quantization.py (hunk 1 above)
Comment (shadeMe) [reply integrations/optimum/src/haystack_integrations/components/embedders/optimum/quantization.py]: Fixed!

**Review by silvanocerza — APPROVED** [review]
Looks good, just two typos to fix and it's good to merge.

**Pull request merged**